internal/controller/stas/containerimagescan_controller.go

```diff
@@ -106,7 +106,7 @@ func (r *ContainerImageScanReconciler) SetupWithManager(mgr ctrl.Manager) error
 				ignoreDeletionPredicate(),
 			)).
 		WithEventFilter(predicate.And(predicates...)).
-		Watches(&source.Channel{Source: r.EventChan}, &handler.EnqueueRequestForObject{}).
+		WatchesRawSource(&source.Channel{Source: r.EventChan}, &handler.EnqueueRequestForObject{}).
```

Comment:
```
// STOP! Consider using For(...), Owns(...), Watches(...), WatchesMetadata(...) instead.
// This method is only exposed for more advanced use cases, most users should use higher level functions.
```
We require _more advanced use cases_, right?

Reply:
Yes, we use the channel to trigger a rescan on time.